test/e2e-techpreview/onclusterlayering_test.go

```diff
@@ -858,16 +858,28 @@ func assertBuildPodIsAsExpected(t *testing.T, cs *framework.ClientSet, mosb *mcf
 	mcoImages, err := ctrlcommon.GetImagesConfig(context.TODO(), cs.GetKubeclient())
 	require.NoError(t, err)
 
-	buildPod, err := cs.CoreV1Interface.Pods(ctrlcommon.MCONamespace).Get(context.TODO(), mosb.Status.BuilderReference.PodImageBuilder.Name, metav1.GetOptions{})
+	buildJob, err := cs.BatchV1Interface.Jobs(ctrlcommon.MCONamespace).Get(context.TODO(), mosb.Status.BuilderReference.PodImageBuilder.Name, metav1.GetOptions{})
 	require.NoError(t, err)
 
+	buildPods, err := cs.CoreV1Interface.Pods(ctrlcommon.MCONamespace).List(context.TODO(), metav1.ListOptions{LabelSelector: fmt.Sprintf("job-name=%s", buildJob.Name)})
+	require.NoError(t, err)
+
+	var buildPodToCheck corev1.Pod
+	if buildPods != nil {
+		for _, pod := range buildPods.Items {
+			if pod.Status.Phase == corev1.PodRunning {
```

Comment:
suggestion: Let's hoist this into a separate helper function which returns the pod for a given job since I see this is also being done elsewhere.

Reply:
Moved to a separate function